### PR TITLE
authors: add validation endpoint

### PR DESCRIPTION
**Pull request by PascalEgn**
* Add schema validation endpoint
* Add schema validation check to create/update call
* Add tests for schema validation endpoint 

ref: https://github.com/cern-sis/issues-inspire/issues/574

### CODE REVIEW

File: backoffice/backoffice/workflows/api/views.py [hunk 1]
```diff
@@ -116,6 +119,14 @@ def create(self, request):
         logger.info("Creating workflow with data: %s", request.data)
         serializer = self.serializer_class(data=request.data)
         if serializer.is_valid(raise_exception=True):
+            logger.info("Validating data against given schema: %s", request.data)
+            # Create request object to pass to schema validation
+            validation_request = request._request
+            validation_request.data = request.data.get("data")
+            validation_response = self.validate(validation_request)
+            if validation_response.status_code != status.HTTP_200_OK:
+                return validation_response
```
Comment (drjova) [inline backoffice/backoffice/workflows/api/views.py]: We can decouple the validation function from the endpoint for better reusability. We can move the validation logic outside, raise validation errors, and simply catch them when needed.

File: backoffice/backoffice/workflows/api/views.py [hunk 2]
```diff
@@ -116,10 +119,14 @@ def create(self, request):
         logger.info("Creating workflow with data: %s", request.data)
         serializer = self.serializer_class(data=request.data)
         if serializer.is_valid(raise_exception=True):
-            workflow = Workflow.objects.create(
-                data=serializer.validated_data["data"],
-                workflow_type=serializer.validated_data["workflow_type"],
-            )
+            logger.info("Validating data against given schema: %s", request.data)
+            validation_errors = list(get_validation_errors(request.data.get("data")))
```
Comment (drjova) [inline backoffice/backoffice/workflows/api/views.py:135]: wouldn't this raise `ValidationError`? :)

Comment (PascalEgn) [reply backoffice/backoffice/workflows/api/views.py:135]: ValidationErrors are no real exceptions, just normal objects, e.g. if there are any, the dont get not risen

Comment (drjova) [reply backoffice/backoffice/workflows/api/views.py:135]: Just to be align with here https://github.com/inspirehep/backoffice/pull/115/files#diff-6c0810e9e873f380de45f0b1ec1906261c199d18654c21bcf3f1efa38aa66ef8R315 that we are catching them

File: backoffice/backoffice/workflows/tests/test_views.py [hunk 3]
```diff
@@ -357,17 +357,17 @@ def test_create_author(self):
...
+                },
+                "_collections": ["Authors"],
+                "$schema": "https://inspirehep.net/schemas/records/authors.json",
             },
         }
 
         url = reverse("api:workflows-authors-list")
         response = self.api_client.post(url, format="json", data=data)
-
+        print(response.json())
```
Comment (DonHaul) [inline backoffice/backoffice/workflows/tests/test_views.py]: remove

File: backoffice/backoffice/workflows/tests/test_views.py [hunk 4]
```diff
@@ -449,8 +449,83 @@ def test_restart_with_params(self):
         response = self.api_client.post(
             url, format="json", data={"params": {"workflow_id": self.workflow.id}}
         )
+        # self.assertContains(response, "test", status_code=200)
```
Comment (DonHaul) [inline backoffice/backoffice/workflows/tests/test_views.py]: remove

**Review by DonHaul — CHANGES_REQUESTED** [review]

**Review by DonHaul — APPROVED** [review]

**Review by drjova — APPROVED** [review]

**Pull request merged**